lib/kioskPrinter.js

```diff
@@ -66,50 +52,75 @@ function printWallet (wallet, printerCfg) {
-                             '^FO50,380' +
-                                 '^ASN,40,35^FDPRIVATE KEY^FS\n' +
-                             '^FO50,420' +
-                                 `^AQN,28,24^FD${wallet.privateAddr.slice(0, wallet.privateAddr.length / 2)}^FS\n` +
-                             '^FO50,450' +
-                                 `^AQN,28,24^FD${wallet.privateAddr.slice(wallet.privateAddr.length / 2)}^FS\n` +
-                             '^FO80,480' +
-                                 `^BQN,2,10,H^FDHM,B0052${wallet.privateAddr}^FS\n` +
-                             '^FO50,940' +
-                                 '^AQN,28,24^FDDon\'t share this QR code with anyone.^FS\n' +
+    openSerialPort(port)
+      .then(() => {
+        const printWalletCmd = '^XA\n' +
+                               '^FO50,250' +
+                               '^ASN,40,35^FDBTC PAPER WALLET^FS\n' +
+                               '^FO50,350' +
+                                   '^AQN,28,24^FDSpend^FS\n' +
+                               '^FO50,380' +
+                                   '^ASN,40,35^FDPRIVATE KEY^FS\n' +
+                               '^FO50,420' +
+                                   `^AQN,28,24^FD${wallet.privateAddr.slice(0, wallet.privateAddr.length / 2)}^FS\n` +
+                               '^FO50,450' +
+                                   `^AQN,28,24^FD${wallet.privateAddr.slice(wallet.privateAddr.length / 2)}^FS\n` +
+                               '^FO80,480' +
+                                   `^BQN,2,10,H^FDHM,B0052${wallet.privateAddr}^FS\n` +
+                               '^FO50,940' +
+                                   '^AQN,28,24^FDDon\'t share this QR code with anyone.^FS\n' +
 
-                             '^FO50,1040' +
-                                 '^AQN,28,24^FDLoad^FS\n' +
-                             '^FO50,1070' +
-                                 '^ASN,40,35^FDPUBLIC KEY^FS\n' +
-                             '^FO50,1110' +
-                                 `^AQN,28,24^FD${wallet.publicAddr}^FS\n` +
-                             '^FO80,1170' +
-                                 `^BQN,2,10,H^FDHM,B0034${wallet.publicAddr}^FS\n` +
+                               '^FO50,1040' +
+                                   '^AQN,28,24^FDLoad^FS\n' +
+                               '^FO50,1070' +
+                                   '^ASN,40,35^FDPUBLIC KEY^FS\n' +
+                               '^FO50,1110' +
+                                   `^AQN,28,24^FD${wallet.publicAddr}^FS\n` +
+                               '^FO80,1170' +
+                                   `^BQN,2,10,H^FDHM,B0034${wallet.publicAddr}^FS\n` +
 
-                             '^CN1\n' +
-                             '^PN0\n' +
-                             '^XZ'
+                               '^CN1\n' +
+                               '^PN0\n' +
+                               '^XZ'
```

Comment:
I'm still not a fan of the + stuff. How about a list of commands that are then joined?

Reply:
I think that's in fact the optimal and sane solution but only when we get into printing the receipts in which we must control the layout. Then we would have a set of functions, each with a very well defined responsibility (printing QRCode, printing logo, printing title, ...) that would be composed to assemble the command that then must be send to the printer.

Reply:
:+1: